File: Basics of Java/Factorial of a number.java
import java.util.*;
class Solution {

public static void main(String args[]) {
       Scanner sc=new Scanner(System.in);
       int n=sc.nextInt();  
       int fact=1;        
       
       if(n==0) System.out.println("1");
       else if(n<0) System.out.println("Error");
       else{
           for(int i=n;i>0;i--) fact=fact*i;
           System.out.println(fact);
       }
    }
}

File: Basics of C++/Factorial of a Number.cpp
#include <iostream>
using namespace std;

int fact(int n) {
   if ((n==0)||(n==1))
   return 1;
   else
   return n*fact(n-1);
}
int main() {
   int n;
   cin>>n;
   if(n<0)
   {
       cout<<"Error"<<endl;
   }
   else
   {
   	cout<<fact(n)<<endl;   
   }
   return 0;
}